packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.tsx

```diff
@@ -214,28 +225,34 @@ const CompleteResetPassword = ({
             );
             break;
           case IntegrationType.OAuth:
-            if (
-              sessionIsVerified &&
-              // only allow this redirect if 2FA is not enabled, otherwise users must enter
-              // their TOTP code first
-              !hasTotp &&
-              // a user can only redirect back to the relier from the original tab
-              // to avoid two tabs redirecting.
-              isOriginalTab()
-            ) {
-              // TODO: this.finishOAuthSignInFlow(account)) in FXA-6518 and possibly
-              // remove the !OAuth check from the React experiment in router.js
-              return;
-            } else if (!isOriginalTab()) {
-              // allows a navigation to a "complete" screen or TOTP screen if it is setup
-              // TODO: check if relier has state
-              if (hasTotp) {
-                // finishing OAuth flow occurs on this page after entering TOTP
-                // TODO: probably need to pass some params
-                hardNavigateToContentServer(
-                  `/signin_totp_code${location.search}`
-                );
-                hardNavigate = true;
+            // allows a navigation to a "complete" screen or TOTP screen if it is setup
+            // TODO: check if relier has state
+            if (hasTotp) {
+              // finishing OAuth flow occurs on this page after entering TOTP
+              // TODO: probably need to pass some params
+              hardNavigateToContentServer(
+                `/signin_totp_code${location.search}`
+              );
+              hardNavigate = true;
+            } else if (sessionIsVerified) {
+              const oauthIntegration = integration as OAuthIntegration;
```

Comment:
Not in your PR but how do you feel about using type guards instead of a `switch` and `integration as OAuthIntegration` cast? I tried the type guards when I first made the integrations and it felt clunky, `switch` felt like the lesser of the two evils at that time but 🤔

Reply:
Type guarrds would be great.  I'll follow up on this.